lua/autorun/client/cl_devtools.lua

```diff
@@ -0,0 +1,133 @@
+local ranOnce = !ranOnce
+if (!ranOnce) then return end -- just to prevent edge cases
+ranOnce = true
+
+local isDev = isDev
+local refreshTime = 2
+local rootFolder = rootFolder or nil
+local fileTimes = fileTimes or {}
+local devReplStr = nil -- classname of all child files, taken from folder
+
+local function replaceFunction(base, regex, replacement)
+	local iStart, iEnd = string.find(base, regex, 1)
+
+	while iStart do
+		local xStart, _ = string.find(base, "(", iEnd, true) -- func name index
+		base = string.sub(base, 1, iStart-1) ..
+			replacement .. "." .. string.sub(base, iEnd + 1, xStart-1) ..
+			" = function" .. string.sub(base, xStart)
+		iStart, iEnd = string.find(base, regex, xStart)
+	end
+	return base
+end
+
+local function refreshFile(path)
+	if (!file.Exists(path, "LUA")) then return print("File " .. path .. " doesn't exist") end
+	if (!isDev) then return end
+	local code = file.Read(path, "LUA")
+
+	-- current filename as class format
+	local curClass = devReplStr or string.TrimRight(string.gsub(path, "^.*/", ""), ".lua")
+
+	-- simple replacement for compatibility, handling weapons and ents, no need for a lexer
+	-- performance impact should be minimal, it's ran once for testing, anyway
+	local storedStr = ".GetStored(\"" .. curClass .. "\")"
+
+	-- replace variables
+	code = string.gsub(string.gsub(code,
+		"%f[%a?]SWEP%.", "weapons" .. storedStr .. "."),
+		"%f[%a?]ENT%.", "scripted_ents" .. storedStr .. ".")
+
+	-- set all function definitions to assign to table
+	code = replaceFunction(replaceFunction(code,
+		"function%s+%f[%a?]SWEP:", "weapons" .. storedStr),
+		"function%s+%f[%a?]ENT:", "scripted_ents" .. storedStr)
+
+	RunString(code, "swampDevTools")
+	print("File refreshed: " .. path)
+end
+
+local function recurseRefresh(times, curPath)
+	local files, folders = file.Find(rootFolder .. curPath .. "*", "LUA")
+	for k, v in pairs(files) do
+		local newtime = file.Time(rootFolder .. curPath .. v, "LUA")
+		if (!times[k]) then -- new file created, or first run
+			times[k] = newtime
+		elseif (newtime != times[k]) then
+			times[k] = newtime
+
+			refreshFile(rootFolder .. curPath .. v)
+		end
+	end
+	for _, v in pairs(folders) do
+		if (!times[v]) then times[v] = {} end
+		if ((rootFolder == "weapons/") or (rootFolder == "entities/")) then
+			devReplStr = v
+		end
+		times[v] = recurseRefresh(times[v], v .. "/")
+	end
+	devReplStr = nil
+	return times
+end
+
+local function timedRefresh()
```

Comment:
you should probably use timer.Create, and have the timer just always be running but return early if !isDev . That is a much more stable way to code things as there is no possibility of multiple versions of the timer running or whatever